pennylane_lightning_kokkos/src/bindings/Bindings.cpp

```diff
@@ -734,6 +742,20 @@ PYBIND11_MODULE(lightning_kokkos_qubit_ops, // NOLINT: No control over
 
     m.def("kokkos_start", []() { Kokkos::initialize(); });
     m.def("kokkos_end", []() { Kokkos::finalize(); });
+
+    py::class_<Kokkos::InitArguments>(m, "InitArguments")
+        .def(py::init<>())
+        .def(py::init<const int &>())
```

Comment:
Should we define a constructor that takes arguments for the other struct variables?
Is there an advantage to defining these as read-write rather than directly setting in the constructor?

Reply:
> Should we define a constructor that takes arguments for the other struct variables?

Yes, we should, especially to deal with the "HIP" and "CUDA" backends.

> Is there an advantage to defining these as read-write rather than directly setting in the constructor?

It prevents having to create an `InitArguments` object for each combination of field values. I think it's convenient to add all functionality in the end. I used read-write to expose the field at the Python layer, so I can write for instance `kokkos_args.num_threads = num_threads`. This is more flexible and convenient in some situations. We could also add setters and getters for each field.